api.js

```diff
@@ -0,0 +1,78 @@
+// API written in express by charmines#1522
+exports.run = async (client) => {
+	const settings = require("./settings.json")
+	const bodyParser = require("body-parser");
+	const moment = require('moment')
+	const erl = require('express-rate-limit')
+	const express = require('express')
+	let app = express()
+
+	// Listen on port in settings
+	app.listen(settings.APIPort, (err) => {
+		if (err) {
+			console.log(err)
+		} else {
+			console.log('Ready at port ' + settings.APIPort)
+		}
+	})
+
+	// Handle requests
+	app.use(bodyParser.json({ type: "application/json" }));
+	app.use(express.urlencoded()) // Decode URLs
+
+	// Ratelimit
+	app.use(erl({
+	  windowMs: 5 * 1000, // 5 Seconds
+	  max: 10 // 10 request per windowMs
+	}))
+
+	// Handle XP API Requests
+	app.get('/xp', async (req, res) => {
+	  let target = await client.guilds.cache.get('731520035717251142').members.cache.get(req.param('userid'))
+	  const profile = await target.settings()
+	  res.json({
+		"username": target.nickname,
```

Comment:
```suggestion
		"username": target.nickname ?? target.user.username,
```

The user's nickname can be `null` when they don't have one.

**BREAKING CHANGE:** Drop support for node v12 or below as this uses v14 syntax.